### DISCUSSION TITLE
OS user `airflow`'s role in `Dockerfile.ci`

### DISCUSSION THREAD

**Discussion by shahar1** [apache/airflow#31714] [Q&A] [answered]
I'm trying to understand Airflow's Dockerfiles better, and I've came across this question -according to breeze [docs](https://github.com/apache/airflow/blob/main/dev/breeze/doc/adr/0006-using-root-user-and-fixing-ownership-for-ci-container.md), the user that runs the container built by `Dockerfile.ci` is the root user. Nonetheless, still we create the `airflow` user and we add it to the sudoers list (as can be seen [in L1242 of Dockerfile.ci](https://github.com/apache/airflow/blob/main/Dockerfile.ci#L1272
)). 
I didn't notice any `USER airflow` instruction within the file (or an equivalent `su airflow` within the CI scripts), so I'm quite convinced that the entire CI is run by the root.
Is there any practical use for the `airflow` user as part of the CI, is it solely for compatibility with related logic in the original image or is it for something else? 

Thanks beforehand!

**Comment by potiuk** [comment, marked as answer]
I think there are (or maybe were) some tests that used it - this might be a remnant of some old behaviour that could be using it. The easiest way to check if is to open a draft PR with removing it and seeing if (and which) things will start failing. Maybe we can just remove, so if the PR succeeds, that would be clear indication we can just merge it.
